lib/text/vtt_text_parser.js

```diff
@@ -415,13 +452,33 @@ shaka.text.VttTextParser = class {
           case 'u':
             nestedCue.textDecoration.push(underline);
             break;
+          default:
+            if (textColorKeys.includes(tag)) {
+              nestedCue.color = textColor[tag];
+            } else if (bgColorKeys.includes(tag)) {
+              nestedCue.backgroundColor = bgColor[tag];
+            }
+            break;
         }
       }
     }
     const isTextNode = shaka.util.XmlUtils.isText(element);
     if (isTextNode) {
-      nestedCue.payload = element.textContent;
-      cues.push(nestedCue);
+      const textArr = element.textContent.split('\n');
```

Comment:
How is this section relevant to default text colors?

Reply:
To enable default text color, a text cue is break into multiple text cues.
Then we assign color to different text cues.
However, line break could not be displayed properly when line break cue is not inserted.
For example, we have two row of text:
```
Each row of roll-up
<c.bg_transparent>    </c>captioning may be set
```
It would be converted to
```
<div style="white-space: pre-wrap; direction: ltr; opacity: 1; justify-content: center; display: flex; flex-flow: row wrap; margin: 0px; align-items: flex-end; font-weight: 400; font-style: normal; position: absolute; top: 69%; float: left; text-align: left; writing-mode: horizontal-tb;">
	<span style="white-space: pre-wrap; direction: ltr; opacity: 1; justify-content: flex-end; display: inline-block; max-width: 100%; font-weight: 400; font-style: normal; text-align: center; writing-mode: horizontal-tb;">Each row of roll-up&nbsp;</span>
	<span style="white-space: pre-wrap; direction: ltr; opacity: 1; justify-content: flex-end; display: inline-block; max-width: 100%; font-weight: 400; font-style: normal; text-align: center; writing-mode: horizontal-tb;">&nbsp;&nbsp;&nbsp;&nbsp;</span>
	<span style="white-space: pre-wrap; direction: ltr; opacity: 1; justify-content: flex-end; display: inline-block; max-width: 100%; font-weight: 400; font-style: normal; text-align: center; writing-mode: horizontal-tb;">captioning may be set&nbsp;</span>
</div>
```
and displayed in 1 line.

Reply:
To enable default text color, a text cue is break into multiple text cues and color is assigned to specific cue.
However, line break could not be displayed properly when line break cue is not inserted.
For example, there are two rows of text:
```
Each row of roll-up
<c.bg_transparent>    </c>captioning may be set
```
It would be converted to
```
<div style="white-space: pre-wrap; direction: ltr; opacity: 1; justify-content: center; display: flex; flex-flow: row wrap; margin: 0px; align-items: flex-end; font-weight: 400; font-style: normal; position: absolute; top: 69%; float: left; text-align: left; writing-mode: horizontal-tb;">
	<span style="white-space: pre-wrap; direction: ltr; opacity: 1; justify-content: flex-end; display: inline-block; max-width: 100%; font-weight: 400; font-style: normal; text-align: center; writing-mode: horizontal-tb;">Each row of roll-up&nbsp;</span>
	<span style="white-space: pre-wrap; direction: ltr; opacity: 1; justify-content: flex-end; display: inline-block; max-width: 100%; font-weight: 400; font-style: normal; text-align: center; writing-mode: horizontal-tb;">&nbsp;&nbsp;&nbsp;&nbsp;</span>
	<span style="white-space: pre-wrap; direction: ltr; opacity: 1; justify-content: flex-end; display: inline-block; max-width: 100%; font-weight: 400; font-style: normal; text-align: center; writing-mode: horizontal-tb;">captioning may be set&nbsp;</span>
</div>
```
and displayed in one row.

Reply:
I see.
Well, could you add a brief comment explaining this, then?

Reply:
Sure. I will add a brief comment in next update, thanks.